std/range/primitives.d

```diff
@@ -2236,7 +2236,7 @@ if (!isNarrowString!(T[]) && !is(T[] == void[]))
 }
 
 /// ditto
-void popFront(C)(ref C[] str) @trusted pure nothrow
+void popFront(C)(scope ref C[] str) @trusted pure nothrow
```

Comment:
Unfortunately, even though `char[]` is implicitly convertible to `const(char)[]`, if I change `ref C[] str` to `ref const(C)[] str` the template will fail to match.